app/uk/gov/hmrc/gform/auth/AuthService.scala

```diff
@@ -67,6 +72,29 @@ class AuthService(
         performAgent(agentAccess, formTemplate, lang, ggAuthorised(RecoverAuthResult.noop), ifSuccessPerformEnrolment)
     }
 
+  private def performAWSALBAuth(formTemplate: FormTemplate, lang: Option[String])(
+    implicit hc: HeaderCarrier): AuthResult = {
+    val encodedJWT: Seq[(String, String)] = hc.otherHeaders.filter(header => header._1 == "x-amzn-oidc-data")
+
+    encodedJWT.headOption match {
+      case None => AuthBlocked("You are not authorized to access this service")
+      case Some((_, jwt)) => {
+        val decoder = Base64.getDecoder
+        val splitJwt = jwt.split("\\.")
+        if (!(jwt.split("\\.").length == 3)) {
+          Logger.error(s"Corrupt JWT received from AWS ALB: [$jwt]")
+          AuthBlocked("You are not authorized to access this service")
+        } else {
+          val payload = new String(decoder.decode(splitJwt(1)))
+          Json.parse(payload) \ "username" match {
+            case JsDefined(value) => AuthSuccessful(AWSALBRetrievals(value.as[String]))
```

Comment:
`value.as[String]` <- What if "username" is not a `String`?

Reply:
It is a string - it's part of the specification

Reply:
`value` is of type `JsValue` (no matter what any specification says) and so `value.as[String]` is unsafe.

Reply:
I evaluate it as an option now